### PR TITLE
PLANNER-2520 VRP quickstart cleanup

### PR DESCRIPTION
**Pull request by yurloc**
Follows up on #156.

CC @dupliaka.

### JIRA
https://issues.redhat.com/browse/PLANNER-2520

### CODE REVIEW

File: use-cases/vehicle-routing/src/main/java/org/acme/vehiclerouting/bootstrap/DemoDataBuilder.java [hunk 1]
```diff
@@ -33,153 +24,161 @@
...
+        if (depotCount < 1) {
+            throw new IllegalStateException(
+                    "Number of depotCount (" + depotCount + ") must be greater than zero.");
         }
 
-        public DemoDataBuilder setDepotCount(int depotCount) {
-                this.depotCount = depotCount;
-                return this;
+        if (northEastCorner.getLatitude() <= southWestCorner.getLatitude()) {
+            throw new IllegalStateException("southWestCorner.getLatitude (" + southWestCorner.getLatitude()
```
Comment (dupliaka) [inline use-cases/vehicle-routing/src/main/java/org/acme/vehiclerouting/bootstrap/DemoDataBuilder.java]: Shouldn't it be 
`must be greater than northEastCorner.getLatitude(" + northEastCorner.getLatitude()`

File: use-cases/vehicle-routing/src/main/java/org/acme/vehiclerouting/bootstrap/DemoDataBuilder.java [hunk 2]
```diff
@@ -33,153 +24,161 @@
...
+                    + ") must be greater than southWestCorner.getLatitude(" + southWestCorner.getLatitude() + ").");
         }
 
-        public DemoDataBuilder setVehicleCapacity(int vehicleCapacity) {
-                this.vehicleCapacity = vehicleCapacity;
-                return this;
+        if (northEastCorner.getLongitude() <= southWestCorner.getLongitude()) {
+            throw new IllegalStateException("southWestCorner.getLongitude (" + southWestCorner.getLongitude()
+                    + ") must be greater than southWestCorner.getLongitude(" + southWestCorner.getLongitude() + ").");
```
Comment (dupliaka) [inline use-cases/vehicle-routing/src/main/java/org/acme/vehiclerouting/bootstrap/DemoDataBuilder.java:132]: Shouldn't it be
`must be greater than northEastCorner.getLongitude(" + northEastCorner.getLongitude() + ").");`

**Review by ge0ffrey — APPROVED** [review]
Looks good to me.

We 'll want a Benchmark class too eventually (if that doesn't exist in the first PR yet) like in the other quickstarts, but first Benchmarks shouldn't be tests, so it's fine to leave that out for now. We'll handle that when we fix the benchmarks.

File: use-cases/vehicle-routing/src/main/java/org/acme/vehiclerouting/bootstrap/DemoDataBuilder.java (hunk 1 above)
Comment (yurloc) [reply use-cases/vehicle-routing/src/main/java/org/acme/vehiclerouting/bootstrap/DemoDataBuilder.java]: Good catch! Thanks, Anna.

Comment (yurloc) [reply use-cases/vehicle-routing/src/main/java/org/acme/vehiclerouting/bootstrap/DemoDataBuilder.java]: North and East coordinates are positive, South and West are negative. So NE corner's latitude and longitude must both be greater than SW's latitude and longitude.

File: use-cases/vehicle-routing/src/main/resources/META-INF/resources/index.html [hunk 3]
```diff
@@ -85,13 +85,14 @@ <h5>Vehicles list</h5>
             <tr>
               <th style="width: 1rem"></th>
               <th>Name</th>
-              <th>Customers</th>
+              <th>Usage</th>
```
Comment (dupliaka) [inline use-cases/vehicle-routing/src/main/resources/META-INF/resources/index.html]: Is there some particular reason to rename it so? 
Or maybe we can specify what does 25/25  mean?

File: use-cases/vehicle-routing/src/main/java/org/acme/vehiclerouting/domain/geo/EuclideanDistanceCalculator.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+public class EuclideanDistanceCalculator implements DistanceCalculator {
+
+    // Approximate Metric Equivalents for Degrees. At the equator for longitude and for latitude anywhere,
+    // the following approximations are valid: 1° = 111 km (or 60 nautical miles) 0.1° = 11.1 km.
+    public static final long METERS_PER_DEGREE = 111_000;
+
+    @Override
+    public long calculateDistance(Location from, Location to) {
+        if (from == to) {
```
Comment (dupliaka) [inline use-cases/vehicle-routing/src/main/java/org/acme/vehiclerouting/domain/geo/EuclideanDistanceCalculator.java]: Don't you want to use equals() here?

File: use-cases/vehicle-routing/src/test/java/org/acme/vehiclerouting/solver/SolverManagerTest.java [hunk 5]
```diff
@@ -16,28 +16,29 @@
...
     @Test
     void solve() throws ExecutionException, InterruptedException {
         VehicleRoutingSolution problem = DemoDataBuilder.builder().setMinDemand(1).setMaxDemand(2)
-                .setVehicleCapacity(15).setCustomerCount(77).setVehicleCount(6).setDepotCount(2)
-                .setSouthWestCorner(new AirLocation(0L, 43.751466, 11.177210))
-                .setNorthEastCorner(new AirLocation(0L, 43.809291, 11.290195)).build();
+                .setVehicleCapacity(25).setCustomerCount(75).setVehicleCount(6).setDepotCount(2)
+                .setSouthWestCorner(new Location(0L, 43.751466, 11.177210))
+                .setNorthEastCorner(new Location(0L, 43.809291, 11.290195)).build();
         solverManager.solve(0L, id -> problem, SolverManagerTest::printSolution).getFinalBestSolution();
```
Comment (dupliaka) [inline use-cases/vehicle-routing/src/test/java/org/acme/vehiclerouting/solver/SolverManagerTest.java:43]: Shell we use here a log?

File: use-cases/vehicle-routing/src/test/java/org/acme/vehiclerouting/bootstrap/DemoDataBuilderTest.java [hunk 6]
```diff
@@ -58,52 +56,58 @@ void correct_builder_builds_ok() {
...
+        assertThatIllegalStateException().isThrownBy(builder::build);
 
         builder = correctBuilder().setMaxDemand(0);
-        assertThrows(IllegalStateException.class, builder::build);
+        assertThatIllegalStateException().isThrownBy(builder::build);
         builder.setMaxDemand(-1);
-        assertThrows(IllegalStateException.class, builder::build);
+        assertThatIllegalStateException().isThrownBy(builder::build);
 
         builder = correctBuilder().setMaxDemand(2);
```
Comment (dupliaka) [inline use-cases/vehicle-routing/src/test/java/org/acme/vehiclerouting/bootstrap/DemoDataBuilderTest.java]: Shell we keep it here? 
We can either add an assignment 
builder = correctBuilder().setMaxDemand(2);
assertDoesNotThrow(builder::build);

or remove it

**Comment by dupliaka** [comment]
How about some rest tests?
https://github.com/yurloc/optaplanner-quickstarts/commit/5008846357b8283cb003c69e8cebb45c25144d89

File: use-cases/vehicle-routing/src/main/resources/META-INF/resources/index.html (hunk 3 above)
Comment (yurloc) [reply use-cases/vehicle-routing/src/main/resources/META-INF/resources/index.html]: It indicates how full the vehicle is. For example, `20/25` means that the vehicle has a capacity of 25 units and the sum of demands of all customers it's visiting is 20.

I agree that "Usage" may not be the best word to describe that. What about "[Load](https://www.merriam-webster.com/dictionary/load)"?

Comment (yurloc) [reply use-cases/vehicle-routing/src/main/resources/META-INF/resources/index.html]: The reason I renamed it from "Customers" is that the column originally showed something like `16/77` meaning that the vehicle goes to 16 customers and there is a total of 77 customers on the map. I didn't consider this an interesting enough piece of information. Also I wouldn't choose a progress bar to display such data.

Anyway, we could show both the load and the number of visits. We just need to find a place for it. I'm afraid another column may not fit in the table.

File: use-cases/vehicle-routing/src/test/java/org/acme/vehiclerouting/bootstrap/DemoDataBuilderTest.java (hunk 6 above)
Comment (yurloc) [reply use-cases/vehicle-routing/src/test/java/org/acme/vehiclerouting/bootstrap/DemoDataBuilderTest.java]: Good catch! I think the intent was to verify an exception is thrown when `minDemand > maxDemand`. I'll fix that.

File: use-cases/vehicle-routing/src/test/java/org/acme/vehiclerouting/solver/SolverManagerTest.java (hunk 5 above)
Comment (yurloc) [reply use-cases/vehicle-routing/src/test/java/org/acme/vehiclerouting/solver/SolverManagerTest.java:43]: No, `sout` is allowed in tests. I find this easier to read without the logging metadata:
```
Vehicle{id=169}: [168, 248, 184, 264, 318, 208, 298, 288, 224, 246, 324, 286, 296, 292, 234, 278, 220, 168]
Vehicle{id=170}: [168, 308, 294, 200, 242, 250, 266, 258, 190, 230, 182, 290, 270, 282, 198, 168]
Vehicle{id=171}: [166, 256, 284, 194, 316, 280, 206, 228, 306, 218, 312, 260, 240, 178, 226, 310, 252, 166]
Vehicle{id=172}: [166, 304, 262, 302, 196, 222, 236, 180, 322, 214, 320, 212, 192, 210, 300, 204, 166]
Vehicle{id=173}: []
Vehicle{id=174}: [168, 274, 276, 188, 238, 314, 176, 216, 254, 202, 186, 272, 244, 268, 232, 168]
```

File: use-cases/vehicle-routing/src/main/java/org/acme/vehiclerouting/domain/geo/EuclideanDistanceCalculator.java (hunk 4 above)
Comment (yurloc) [reply use-cases/vehicle-routing/src/main/java/org/acme/vehiclerouting/domain/geo/EuclideanDistanceCalculator.java]: Yes, that follows the best practice.

File: use-cases/vehicle-routing/src/main/resources/META-INF/resources/index.html (hunk 3 above)
Comment (dupliaka) [reply use-cases/vehicle-routing/src/main/resources/META-INF/resources/index.html]: In my opinion the perfect column header that could explain the bar is **Load (dem/cus)** to identify who is who

File: use-cases/vehicle-routing/src/test/java/org/acme/vehiclerouting/solver/SolverManagerTest.java (hunk 5 above)
Comment (dupliaka) [reply use-cases/vehicle-routing/src/test/java/org/acme/vehiclerouting/solver/SolverManagerTest.java:43]: OK

**Review by dupliaka — APPROVED** [review]
LGTM

**Pull request merged**